bigquery_etl/cli/query.py

```diff
@@ -702,61 +717,54 @@ def backfill(
-
     for query_file in query_files:
         query_file_path = Path(query_file)
 
-        depends_on_past = False
-        date_partition_parameter = "submission_date"
-
         try:
             metadata = Metadata.of_query_file(str(query_file_path))
-            depends_on_past = metadata.scheduling.get(
-                "depends_on_past", depends_on_past
-            )
-            date_partition_parameter = metadata.scheduling.get(
-                "date_partition_parameter", date_partition_parameter
-            )
+        except FileNotFoundError:
+            click.echo(f"Can't run backfill without metadata for {query_file_path}.")
+            continue
 
-            # For backwards compatibility assume partitioning type is day
-            # in case metadata is missing
-            if metadata.bigquery:
-                partitioning_type = metadata.bigquery.time_partitioning.type
-            else:
-                partitioning_type = PartitionType.DAY
-                click.echo(
-                    "Bigquery partitioning type not set. Using PartitionType.DAY"
-                )
+        depends_on_past = metadata.scheduling.get("depends_on_past", False)
+        # If date_partition_parameter isn't set it's assumed to be submission_date:
+        # https://github.com/mozilla/telemetry-airflow/blob/dbc2782fa23a34ae8268e7788f9621089ac71def/utils/gcp.py#L194C48-L194C48
+        date_partition_parameter = metadata.scheduling.get(
+            "date_partition_parameter", "submission_date"
+        )
+        date_partition_offset = metadata.scheduling.get("date_partition_offset", 0)
 
-            match partitioning_type:
-                case PartitionType.DAY:
-                    dates = [
-                        start_date + timedelta(i)
-                        for i in range((end_date - start_date).days + 1)
-                    ]
-                case PartitionType.MONTH:
-                    dates = list(
-                        rrule(
-                            freq=MONTHLY,
-                            dtstart=start_date.replace(day=1),
-                            until=end_date,
-                        )
-                    )
-                    # Dates in excluded must be the first day of the month to match `dates`
-                    exclude = [
-                        date.fromisoformat(day).replace(day=1).strftime("%Y-%m-%d")
-                        for day in exclude
-                    ]
-                case _:
-                    raise ValueError(
-                        f"Unsupported partitioning type: {partitioning_type}"
+        # For backwards compatibility assume partitioning type is day
+        # in case metadata is missing
+        if metadata.bigquery:
+            partitioning_type = metadata.bigquery.time_partitioning.type
+        else:
+            partitioning_type = PartitionType.DAY
+            click.echo("Bigquery partitioning type not set. Using PartitionType.DAY")
```

Comment:
Will change this assumption in upcoming PR(s), for example backfilling unpartitioned tables. See - https://github.com/mozilla/bigquery-etl/issues/4532 (though I think this is somewhat supported via `--no-partition`)